src/main/java/seedu/address/model/person/DateOfLastVisit.java

```diff
@@ -44,6 +50,28 @@ public static boolean isValidDateOfLastVisit(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Compares two dates of last visit by year then month then day to find
+     * the right order.
+     *
+     * @param other the date to be compared to.
+     * @return an integer which specifies which date is earlier or if they are equal.
+     */
+    @Override
+    public int compareTo(DateOfLastVisit other) {
+        int yearComparison = this.dayMonthYear[2].compareTo(other.dayMonthYear[2]);
+        if (yearComparison == 0) {
+            int monthComparsion = this.dayMonthYear[1].compareTo(other.dayMonthYear[1]);
+            if (monthComparsion == 0) {
+                return this.dayMonthYear[0].compareTo(other.dayMonthYear[0]);
+            } else {
+                return monthComparsion;
+            }
```

Comment:
a minor issue, perhaps you could consider making this small portion not use arrowhead style. But if its too troublesome to change then i guess its fine too

Reply:
I know it looks like an arrowhead but its a single nested if statement. This is the best I could think of, unless you have another suggestion. Given what the code is trying to do, there doesn't seem to be a way to avoid the nesting.